browser/ipfs/ipfs_tab_helper.cc

```diff
@@ -107,18 +114,23 @@ void IPFSTabHelper::IPFSLinkResolved(const GURL& ipfs) {
   UpdateLocationBar();
 }
 
-void IPFSTabHelper::HostResolvedCallback(const std::string& host,
-                                         const std::string& dnslink) {
+void IPFSTabHelper::HostResolvedCallback(
+    absl::optional<std::string> x_ipfs_path_header,
+    const std::string& host,
+    const absl::optional<std::string>& dnslink) {
   GURL current = web_contents()->GetURL();
+
   if (current.host() != host || !current.SchemeIsHTTPOrHTTPS())
     return;
-  if (dnslink.empty())
+  if (!dnslink || dnslink.value().empty()) {
+    if (x_ipfs_path_header) {
+      XIPFSPathLinkResolved(ResolveXIPFSPathUrl(x_ipfs_path_header.value()));
```

Comment:
Shouldn't we also check before calling `XIPFSPathLinkResolved`  if url returned by `ResolveXIPFSPathUrl` is valid as we do for value return from `ResolveDNSLinkUrl`?

Reply:
Fixed